packages/fuel-indexer-metrics/src/lib.rs

```diff
@@ -1,31 +1,54 @@
 #![deny(unused_crate_dependencies)]
 
 pub mod core;
-pub mod queries;
+pub mod db;
 
 use crate::core::{Metric, Metrics};
+use axum::{
+    body::Body,
+    http::StatusCode,
+    response::{IntoResponse, Response},
+};
 use lazy_static::lazy_static;
-use prometheus::{self, Encoder, Error, TextEncoder};
-use thiserror::Error;
+use prometheus::{self, Encoder, TextEncoder};
+use prometheus_client::encoding::text::encode;
 
 lazy_static! {
     pub static ref METRICS: Metrics = Metrics::init();
 }
 
-pub(crate) type MetricsResult<T> = std::result::Result<T, MetricsError>;
-
-#[derive(Error, Debug)]
-pub enum MetricsError {
-    #[error("Encoding metrics error: {0:#?}")]
-    MetricsEncoding(#[from] Error),
-}
-
-pub fn encode_metrics_response() -> MetricsResult<(Vec<u8>, String)> {
-    let mut buff = Vec::new();
+pub fn encode_metrics_response() -> impl IntoResponse {
+    let mut encoded_families = Vec::new();
     let encoder = TextEncoder::new();
+
     let metric_families = prometheus::gather();
-    match encoder.encode(&metric_families, &mut buff) {
-        Ok(_) => Ok((buff, encoder.format_type().to_owned())),
-        Err(e) => Err(MetricsError::MetricsEncoding(e)),
+
+    if encoder
+        .encode(&metric_families, &mut encoded_families)
+        .is_err()
+    {
+        return error_body();
     }
+
+    let mut encoded = String::from_utf8_lossy(&encoded_families).into_owned();
+
+    if encode(&mut encoded, &METRICS.web.registry).is_err() {
+        return error_body();
+    }
+
+    if encode(&mut encoded, &METRICS.db.postgres.registry).is_err() {
+        return error_body();
+    }
+
+    Response::builder()
+        .status(StatusCode::OK)
+        .body(Body::from(encoded))
+        .unwrap()
+}
```

Comment:
I wonder if it's possible to use something like [`.and_then(..)`](https://doc.rust-lang.org/std/result/enum.Result.html#method.and_then) here.